Update compatibility matrix with the actual release data instead of examples (#349)
vmware-tanzu/tanzu-cli: docs/full/policy.md

  - For example, the below table shows the compatible Tanzu Plugin Runtime versions with different releases of Tanzu CLI.
  - ***Note***: In the below table v0.11, 0.25, 0.28, 0.29 versions are `legacy` library versions from Tanzu-Framework and `x` is any existing or future patch version.

    | Tanzu CLI Version | Compatible Tanzu Plugin Runtime Versions |
    |-------------------|------------------------------------------|
    | v0.90.x           | v0.11.x                                  |
    |                   | v0.25.x                                  |
    |                   | v0.28.x                                  |
    |                   | v0.29.x                                  |
    |                   | v0.90.x                                  |

- ***Important***: Based on this compatibility policy, the user can always upgrade the Tanzu CLI to the latest version without worrying about existing plugin compatibility.
- This means any change in the contract between Tanzu CLI and Tanzu Plugin Runtime must be done in a backward-compatible manner.
- ***FAQ***:

>   - What happens if I upgrade to a newer version of CLI v0.90.0 and use a plugin developed with Plugin Runtime v0.28.0?

Reviewer: ```suggestion
  - What happens if I upgrade to the CLI v0.90.0 and use a plugin developed with Plugin Runtime v0.28.0?
```

>     - As CLI is always compatible with all existing plugins, plugins developed with Plugin Runtime v0.28.0 should continue to work.

Reviewer: ```suggestion
    - As the CLI is always compatible with all existing plugins, plugins developed with Plugin Runtime v0.28.0 will continue to work.
```

>   - If I have developed a plugin with (v0.11, 0.25, 0.28, 0.29) of Tanzu Plugin Runtime (from Tanzu-Framework repository), Will my plugins be compatible with v0.90 Tanzu CLI?

Reviewer: ```suggestion
  - If I have developed a plugin with (v0.11, 0.25, 0.28, 0.29) of Tanzu Plugin Runtime (from the Tanzu-Framework repository), will my plugins be compatible with v0.90 Tanzu CLI?
```

    - Yes. As shown in the above diagram and the backward compatibility guarantees provided with the v0.90 release of Tanzu CLI, all plugins developed with (v0.11, 0.25, 0.28, 0.29) of Tanzu Plugin Runtime will be compatible with v0.90 Tanzu CLI.
  - When will my plugin be incompatible with the CLI?
    - The user may be using a CLI version that is old and some new functionality implemented in a plugin with newer Plugin Runtime is not compatible with the old CLI. However, the old CLI will still be able to invoke the plugin with all the compatible features. Also, upgrading CLI to a newer version should resolve this new feature incompatibility issue.

>     - Example: If a plugin is developed with Tanzu Plugin Runtime v1.3.1 which introduced a new feature X which works in combination with a newer version of Tanzu CLI then if the user is using old Tanzu CLI v1.2.0 (which doesn’t support this new feature X introduced in Plugin Runtime v1.3.1) while invoking a plugin, that new functionality might not work when using old CLI. However, other functionalities of that plugin will continue to work as expected with the old CLI. This will make the new feature in the plugin incompatible with the installed CLI. However, the user can upgrade the CLI to the latest v1.3.0 version to make the new feature compatible again. This type of situation will always be communicated with the above table when this situation arises.

Reviewer: ```suggestion
    - Example: If a plugin is developed with Tanzu Plugin Runtime v1.3.1 which introduced a new feature X which works in combination with a newer version of Tanzu CLI then if the user is using an old Tanzu CLI v1.2.0 (which doesn’t support this new feature X introduced in Plugin Runtime v1.3.1) while invoking a plugin, that new functionality might not work when using the old CLI. However, other functionalities of that plugin will continue to work as expected with the old CLI. This will make the new feature in the plugin incompatible with the installed CLI. However, the user can upgrade the CLI to the latest v1.3.0 version to make the new feature compatible again. This type of situation will always be communicated with the above table when this situation arises.
```
